test/tap/tests/test_admin_stats-t.cpp

```diff
@@ -0,0 +1,180 @@
+/**
+ * @file test_admin_stats-t.cpp
+ * @brief This tests the Statistics module and its lookup code and tables.
+ * @date 2021-10-28
+ */
+
+#include <algorithm>
+#include <string>
+#include <stdio.h>
+#include <unistd.h>
+#include <vector>
+
+#include <mysql.h>
+#include <mysql/mysqld_error.h>
+
+#include "tap.h"
+#include "command_line.h"
+#include "utils.h"
+
+using std::string;
+
+int main(int argc, char** argv) {
+	CommandLine cl;
+
+	if (cl.getEnv()) {
+		diag("Failed to get the required environmental variables.");
+		return -1;
+	}
+
+    /** @brief Minimum number of distinct variable_name strings in the history_mysql_status_variables_lookup table */
+    const int min_distinct_variable_names = 50;
+
+	plan(5);
+
+	MYSQL* proxysql_admin = mysql_init(NULL);
+
+	// Initialize connections
+	if (!proxysql_admin) {
+		fprintf(stderr, "File %s, line %d, Error: %s\n", __FILE__, __LINE__, mysql_error(proxysql_admin));
+		return -1;
+	}
+
+	// Connnect to local proxysql
+	if (!mysql_real_connect(proxysql_admin, cl.host, cl.admin_username, cl.admin_password, NULL, cl.admin_port, NULL, 0)) {
+		fprintf(stderr, "File %s, line %d, Error: %s\n", __FILE__, __LINE__, mysql_error(proxysql_admin));
+		return -1;
+	}
+
+	// Setup the interval of how often new status entries are created
+	uint16_t new_stats_interval_sec = 2;
+
+	// @TODO: run command to set the interval
+
+	// If on a fresh install, wait long enough for the first run of stats to be created
+	// The lookup table will be empty until the first run!
+	sleep(new_stats_interval_sec * 2);
```

Comment:
Maybe a little too much? `new_stats_interval_sec + 1` should be just fine. The valid intervals for being set are documented here: https://proxysql.com/documentation/global-variables/admin-variables/#admin-stats_mysql_connections. So something like the following will do the trick:

```
	// @TODO: run command to set the interval
	MYSQL_QUERY(proxysql_admin, "SET admin-stats_mysql_connections=5");
	MYSQL_QUERY(proxysql_admin, "LOAD ADMIN VARIABLES TO RUNTIME");
```

Reply:
Thanks! Tentative, I wasn't sure what would be appropriate here to ensure enough time to populate initially. Updating...